### ISSUE TITLE
Request: validate buildin types imported from typing

### ISSUE THREAD

**Issue by Goldziher** [snok/flake8-type-checking#129]
Hi, 

So in a project I see someone importing types from collections:

```python

from collections.abc import Generator

...

```

But this is also imported and exported by typing, which is the more correct import in this case - because this is used for typing only. 

So my suggesiton would be to ensure that all typings are imported from typings.

**Comment by sondrelg** [comment]
I've thought this too, but check out [PEP585](https://peps.python.org/pep-0585/) which added generics syntax back to the collections module. I could be wrong, but my current view is that it's better to prefer collections over typing where possible (3.7 with futures imports, or 3.9 w/o futures imports). This way the type hint refers to the actual resource, and not just a concept.

**Comment by sondrelg** [comment]
I also like typing more, since I'm more familiar with it 😄 but this seems like the correct take. Also think `pyupgrade` is probably the best way of breaking (or at least correcting for) the habit 👍
